java-checks/src/main/java/org/sonar/java/checks/CollectionInappropriateCallsCheck.java

```diff
@@ -100,6 +100,10 @@ private static Type getTypeParameter(Type collectionType) {
   }
 
   private static boolean isArgumentCompatible(Type argumentType, Type collectionParameterType) {
+    // FIXME SONARJAVA-1514 subtyping of type using generics is not handled correctly, but wildcards are covered
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1134)